includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -420,216 +576,80 @@ public function getData() {
 				]
 			];
 		}
-
-		return $data;
 	}
 
-	private function getLicenseData() {
-		$licenseText = WikiFactory::getVarByName( 'wgRightsText', $this->wikiId )->cv_value ?: $this->wg->RightsText;
+	private function setSectionData( $sectionName, $sectionBaseData ) {
+		$sectionData = $this->getSectionData( $sectionBaseData );
 
-		return [
-			'type' => 'link-text',
-			'title' => [
-				'type' => 'text',
-				'value' => $licenseText
-			],
-			'href' => $this->getLicenseUrl()
-		];
+		if ( !empty( $sectionData ) ) {
+			$this->data[$sectionName] = $sectionData;
+		}
 	}
 
-	private function getFandomOverview() {
-		$out = [
-			'links' => [ ]
-		];
+	private function getSectionData( $sectionBaseData ) {
+		$sectionData = [ ];
 
-		if ( $this->lang === self::DEFAULT_LANG ) {
-			$out['links'] = [
-				[
-					'type' => 'link-branded',
-					'brand' => 'games',
-					'title' => [
-						'type' => 'translatable-text',
-						'key' => 'global-footer-fandom-link-vertical-games'
-					],
-					'href' => 'http://fandom.wikia.com/games'
-				],
-				[
-					'type' => 'link-branded',
-					'brand' => 'movies',
-					'title' => [
-						'type' => 'translatable-text',
-						'key' => 'global-footer-fandom-link-vertical-movies'
-					],
-					'href' => 'http://fandom.wikia.com/movies'
-				],
-				[
-					'type' => 'link-branded',
-					'brand' => 'tv',
-					'title' => [
-						'type' => 'translatable-text',
-						'key' => 'global-footer-fandom-link-vertical-tv'
-					],
-					'href' => 'http://fandom.wikia.com/tv'
-				],
-			];
+		if ( !empty( $sectionBaseData['header'] ) ) {
+			$sectionData['header'] = $sectionBaseData['header'];
 		}
 
-		$out['links'][] = [
-			'type' => 'link-branded',
-			'brand' => 'fan-communities',
-			'title' => [
-				'type' => 'translatable-text',
-				'key' => 'global-footer-fandom-link-fan-communities'
-			],
-			'href' => $this->getHref( 'fan-communities' )
-		];
-
-		return $out;
-	}
+		if ( !empty( $sectionBaseData['description'] ) ) {
+			$sectionData['description'] = $sectionBaseData['description'];
```

Comment:
Do we need these two parts for the header and description since they don't modify `$sectionBaseData`? This method could in theory just be something like:
```php
	private function getSectionData( $sectionBaseData ) {
		if ( !empty( $sectionBaseData['links'] ) ) {
			$linksData = $this->getLinksData( $sectionBaseData['links'] );
			if ( !empty( $linksData ) ) {
				$sectionBaseData['links'] = $linksData;
			} else {
				unset( $sectionBaseData['links'] );
			}
		}

		return $sectionBaseData;
	}
```